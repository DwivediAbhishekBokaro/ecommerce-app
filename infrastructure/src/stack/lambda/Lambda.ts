import { Duration } from "aws-cdk-lib";
import { Runtime, Tracing, Function, Code } from "aws-cdk-lib/aws-lambda";
import { Construct } from "constructs";

export function createLambdaFunctionUsing(scope: Construct, deploymentStageName: string){

    return new Function(scope, 'MyEcommerceLambda', {
        functionName: `my-ecommerce-lambda-${deploymentStageName}`,
        runtime: Runtime.JAVA_17,
        code: Code.fromAsset('../code/dist'), // Path to your lambda code
        handler: '',
        timeout: Duration.seconds(30),
        memorySize: 128,
        environment: {
            STAGE: deploymentStageName,
            // Add other environment variables as needed
        },
        tracing: Tracing.ACTIVE,
    });
}