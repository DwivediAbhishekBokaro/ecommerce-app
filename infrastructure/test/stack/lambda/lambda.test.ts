import { Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { createLambdaFunctionUsing } from '../../../src/stack/lambda/Lambda';

describe('Lambda Infrastructure', () => {
  it('should create Lambda function with correct properties', () => {
    const stack = new Stack();

    createLambdaFunctionUsing(stack, 'testStage');

    const template = Template.fromStack(stack);

    template.hasResourceProperties('AWS::Lambda::Function', {
      FunctionName: 'my-ecommerce-lambda-testStage',
      Runtime: 'java17',
      Timeout: 30,
      MemorySize: 128,
      TracingConfig: {
        Mode: 'Active',
      },
      Environment: {
        Variables: {
          STAGE: 'testStage',
        },
      },
    });
  });
});
