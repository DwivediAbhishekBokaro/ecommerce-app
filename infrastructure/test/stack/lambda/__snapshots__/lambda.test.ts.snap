// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create lambda function with correct properties 1`] = `
{
  "Resources": {
    "MyEcommerceLambda1AC01721": {
      "DependsOn": [
        "MyEcommerceLambdaServiceRoleDefaultPolicy13D9A0C6",
        "MyEcommerceLambdaServiceRole809E1CF8",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "1ebc9d3ac2033816c4abb63e4afd69d350b4aba8704cc9236b82ea520b74f4b0.zip",
        },
        "Environment": {
          "Variables": {
            "STAGE": "testStage",
          },
        },
        "FunctionName": "my-ecommerce-lambda-testStage",
        "Handler": "",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "MyEcommerceLambdaServiceRole809E1CF8",
            "Arn",
          ],
        },
        "Runtime": "java17",
        "Timeout": 30,
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "MyEcommerceLambdaServiceRole809E1CF8": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MyEcommerceLambdaServiceRoleDefaultPolicy13D9A0C6": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "MyEcommerceLambdaServiceRoleDefaultPolicy13D9A0C6",
        "Roles": [
          {
            "Ref": "MyEcommerceLambdaServiceRole809E1CF8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
